test/bmv2_parsing_check.bzl

```diff
@@ -0,0 +1,67 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+def _subset_diff_impl(ctx):
+    # diff expected with tmp file
+    ctx.actions.write(
+        output = ctx.outputs.executable,
+        content = "{binary} < {input} | python {py_file} {input}".format(
```

Comment:
I find it a bit confusing that you use abstract names "binary" and "py_file" here when we only every call this with very concrete inputs.